### ISSUE TITLE
Refactor huge files into more manageable pieces

### ISSUE THREAD

**Issue by FrNecas** [teemtee/tmt#1082]
Some of our files have grown quite large over the course of time, mainly `base.py`, `utils.py` and perhaps even `cli.py` could be considered. At least for me, it is getting harder to navigate and find what I am looking for. Some of these could be nicely refactored into smaller files, e.g. with only a single class (or in the case of `cli.py` maybe a file for each command group could be useful). In some places we could also simplify the classes themselves a bit. This is IMHO a bit of a technical debt that we should somehow address.

One of the problems I see is the change of API. I remember someone some time ago mentioned that some classes from tmt are used in other project as imports (I think it was in `base.py`) so this would break the usage (so maybe should be done for 2.0)

@psss @lukaszachy thoughts?

**Comment by psss** [comment]
Good idea. Definitely makes sense to me to place some stuff into separate files. And this does not have to break python API. It would be just about better organizing stuff across individual files. For example `tmt.utils.StructuredField` could go to a separate file and would be imported back to `utils`. Having one class per file could also make sense, especially for the large ones like `base`. Before doing this, however, I'd suggest to process as many open pull requests as possible to prevent unnecessary merge conflicts.

**Comment by psss** [comment]
Some related work already done or in progress, tracked on different places. Shall we still keep this one?

**Comment by psss** [comment]
> Some related work already done or in progress, tracked on different places. Shall we still keep this one?

Let's use this one as an umbrella issue for tracking issues & pull requests related to improvements in this area.

**Comment by psss** [comment]
Three items have been proposed for the housekeeping sprint:

* #4326 
* #4325 
* #4529
